android/pytorch_android/build.gradle

```diff
@@ -79,4 +79,49 @@ task sourcesJar(type: Jar) {
     classifier = 'sources'
 }
 
+def getLibtorchHeadersDir() {
+  def abi = ABI_FILTERS.split(",")[0]
+  return "$rootDir/../build_android_$abi/include";
+}
+
+afterEvaluate {
+  if (POM_PACKAGING == 'aar') {
+    android.libraryVariants.all { variant ->
+      variant.outputs.each { output ->
+        File f = output.outputFile
+        if (f.name.endsWith(".aar")) {
+          output.assemble.finalizedBy addFolderToAarTask(
+              "addHeadersToAar" + variant.name,
+              f.path.replaceAll(variant.name, VERSION_NAME),
+              getLibtorchHeadersDir(),
+              "headers")
+        }
+      }
+    }
+  }
+}
+
+def addFolderToAarTask(taskName, aarPath, folderPath, folderPathInAar) {
```

Comment:
just curious: can we include header files before packing AAR file instead of unpacking/repacking here?

Reply:
I tried approach for including headers during packing of aar, it was very hacky and dependent on android gradle plugin internal structure, that may change version to version. I decided that repacking aar will be more stable for updating android gradle plugin versions.